File: Biseksi.m
% Numerical Method
% Yesaya Ananda Djaya
% 5002211156

% Metoda Biseksi

clear 
clc 

batas_iterasi = 20;
iterasi = 0;
batas_galat = 10^-5;

a = 1;
b = 2;

plama = 0;

while iterasi < batas_iterasi
    iterasi = iterasi + 1;

    fprintf('Iterasi ke-%d\n', iterasi)
    fprintf('a = %.6f\n',a)
    fprintf('b = %.6f\n',b)

    fa = a^4 - 8*a^3 + 23*a^2 + 16*a - 50;
    fb = b^4 - 8*b^3 + 23*b^2 + 16*b - 50;

    p = (a+b)/2;

    fprintf('p = %.6f\n', p)

    fp = p^4 - 8*p^3 + 23*p^2 + 16*p - 50;

    fprintf('f(a) = %.6f\n', fa)
    fprintf('f(b) = %.6f\n', fb)
    fprintf('f(p) = %.6f\n', fp)
    
    if fa * fp > 0
        a = p;
        fprintf('p berubah menjadi a baru \n')
    else
        b = p;
        fprintf('p berubah menjadi b baru \n')
    end

    if iterasi > 1
        galat = abs((p - plama) / p) * 100;
        fprintf('Galat = %.6f%%\n', galat)

        if galat < batas_galat * 100
            fprintf('Galat sudah di bawah batas. Proses berhenti.\n')
            break
        end
    end

    plama = p;
    
    fprintf('\n')
end

File: Biseksi.py
# Numerical Method
# Yesaya Ananda Djaya
# 5002211156

# Metode Biseksi

batas_iterasi = 20
iterasi = 0

a = 1
b = 2

fa = -2*a**2 + (5*a) - 10 
fb = -2*b**2 + (5*b) - 10 


print(f'f(a) = {fa}')
print(f'f(b) = {fb}') 

while iterasi < batas_iterasi:
    iterasi += 1

    p = (a + b) / 2 

    print(f'p = {p}')

    
    fp = -2*p**2 + (5*p) - 10
    print(f'f(p) = {fp}')
    print()


    if fa * fp > 0:
        a = p
        print(f'a(baru) = {a}')
        print(f'b(baru) = {b}')
        fa = fp
    else:
        b = p
        print(f'a(baru) = {a}')
        print(f'b(baru) = {b}')
        fb = fp

# Selesai